pkg/phlaredb/phlaredb.go

```diff
@@ -273,6 +152,9 @@ func (f *PhlareDB) loop() {
 			f.runBlockQuerierSync(ctx)
 		case <-blockScanTicker.C:
 			f.runBlockQuerierSync(ctx)
+		case b := <-f.evictCh:
+			b.evicted, b.err = f.blockQuerier.evict(b.blockID)
+			close(b.done)
```

Comment:
Not related to the PR, but: to me it's not entirely clear whether the head block is visible to queries in-between `Flush` and `runBlockQuerierSync`:
```go
		case <-f.Head().flushCh:
			if err := f.Flush(ctx); err != nil {
				level.Error(f.logger).Log("msg", "flushing head block failed", "err", err)
				continue
			}
			f.runBlockQuerierSync(ctx)
```